face/UnifiedFaceCollection/python/unified_face_collection.py

```diff
@@ -0,0 +1,346 @@
+        face_list_response = requests.get(face_list_url, headers=self.headers)
+        if face_list_response.status_code != 200:
+            return False
+
+        face_data = face_list_response.json()
+        user_data = face_data.get("userData")
+        if not user_data:
+            return False
+
+        user_data_dict = json.loads(user_data)
+        person_id = user_data_dict.get("personId")
+        person_persisted_face_id = user_data_dict.get("personPersistedFaceId")
+
+        if person_id and person_persisted_face_id:
+            # Remove face from Large Person Group
+            person_face_url = f"{self.face_api_url}/largepersongroups/{self.large_person_group_id}/persons/{person_id}/persistedfaces/{person_persisted_face_id}"
+            person_face_response = requests.delete(person_face_url, headers=self.headers)
+            if person_face_response.status_code != 200:
+                return False
+
+        # Remove face from Large Face List
+        face_list_delete_response = requests.delete(face_list_url, headers=self.headers)
+        if face_list_delete_response.status_code != 200:
+            return False
+
+        return True
+
+    def remove_person(self, person_identifier):
+        # Determine if the identifier is a name or an ID
+        person_id = None
+        if isinstance(person_identifier, str) and len(person_identifier) == 36:
+            person_id = person_identifier
+        else:
+            person = self.get_person_by_name(person_identifier)
+            if person:
+                person_id = person['personId']
+            else:
+                return False
+
+        # Retrieve the person's faces from the person group
+        person_faces_url = f"{self.face_api_url}/largepersongroups/{self.large_person_group_id}/persons/{person_id}"
+        person_faces_response = requests.get(person_faces_url, headers=self.headers)
+        if person_faces_response.status_code != 200:
+            return False
+
+        person_faces = person_faces_response.json()
+
+        # Update userData in the Large Face List
+        for person_persisted_face_id in person_faces['persistedFaceIds']:
+            # Retrieve userData from the face in the person group
+            person_face_url = f"{self.face_api_url}/largepersongroups/{self.large_person_group_id}/persons/{person_id}/persistedfaces/{person_persisted_face_id}"
+            person_face_response = requests.get(person_face_url, headers=self.headers)
+            if person_face_response.status_code != 200:
+                continue
+
+            face_data = person_face_response.json()
+            user_data = face_data.get("userData")
+            if user_data:
+                user_data_dict = json.loads(user_data)
+                persisted_face_id = user_data_dict.get("persistedFaceId")
+
+                # Update userData for the face in the Large Face List
```

Comment:
Does it make more sense to clean up the records associated with the person name in face list?

Reply:
Imaging someone who uses the unified wrapper for person:face=1:1 scenario. He might only call `add_face(image, "PersonName")`, `remove_person("PersonName")` and `find_face(image, search_type='person')`. After thousands of iterations, there will be a lot of unwanted stuff left in the internal face list.

Reply:
> Imaging someone who uses the unified wrapper for person:face=1:1 scenario. He might only call `add_face(image, "PersonName")`, `remove_person("PersonName")` and `find_face(image, search_type='person')`. After thousands of iterations, there will be a lot of unwanted stuff left in the internal face list.

Add an option to allow delete associated faces in remove_person function.

Reply:
> Does it make more sense to clean up the records associated with the person name in face list?

We do not save the person name in the face list.